shopelectro/views/catalog.py

```diff
@@ -25,6 +26,62 @@ def get_products_count(request):
     return PRODUCTS_ON_PAGE_MOB if mobile_view else PRODUCTS_ON_PAGE_PC
 
 
+def get_view_type(request):
+    view_type = request.session.get('view_type', 'tile')
+    assert view_type in ['list', 'tile']
+    return view_type
+
+
+def get_catalog_context(request, category, raw_tags, page_number, per_page, sorting_index):
+    all_tags = newcontext.Tags(models.Tag.objects.all())
+
+    if raw_tags:
+        selected_tags = newcontext.tags.Checked404Tags(
+            newcontext.tags.ParsedTags(
```

Comment:
we can always pass `raw_tags` (or `uri_tags`) to Tags context.
So, ParsedTags will not be needed

Reply:
`ParsedTags` extract tag slugs from `raw_tags` and find them in db, but it seems the `else:` branch is excess, so i will remove it